frame/ethereum/src/lib.rs

```diff
@@ -241,7 +241,7 @@ impl<T: Trait> Module<T> {
 
 		let ommers = Vec::<ethereum::Header>::new();
 		let header = ethereum::Header {
-			parent_hash: frame_system::Module::<T>::parent_hash(),
+			parent_hash: Self::current_block_hash().unwrap_or_default(),
```

Comment:
A note to reviewers, that before `on_finalize`, `CurrentBlock` actually holds the block built of the previous block. Maybe we should consider renaming it to something more appropriate to avoid confusions.